src/__tests__/discriminatedUnions.test.ts

```diff
@@ -144,6 +193,241 @@ test("wrong schema - duplicate discriminator values", () => {
   }
 });
 
+test("valid - `DiscriminatedUnion` as a union option", () => {
+  const A = z.object({ type: z.literal("a"), a: z.literal(1) });
+  const B = z.object({ type: z.literal("b"), b: z.literal(2) });
+  const C = z.object({ type: z.literal("c").optional(), c: z.literal(true) });
+  const AorB = z.discriminatedUnion("type", [A, B]);
+  const schema = z.discriminatedUnion("type", [AorB, C]);
```

Comment:
Would it make sense to either:
- not allow adding discriminated unions with same discriminator as option, but add a merge function instead or
- in the implementation of create flatten the structure and add the individual sub-schema's to the options if the discriminator is the same

Reply:
Oh great idea @teunmooij! I suspect it might be easier on users to just have this happen behind the scenes on `create()`. Let me add that functionality.

Reply:
@teunmooij see [latest commit](https://github.com/colinhacks/zod/pull/1589/commits/0f16bf71779e02a94fe81b5dc7648834cc6522ab)

Reply:
@maxArturo , LGTM. Nice improvement. (if it was up to me I'd pull your branch to double check and then merge it, would be nice to have this functionality in Zod. It's taking a very very long time for some reason)